pkg/verify/signed_entity.go

```diff
@@ -172,10 +176,39 @@ func NewVerificationResult() *VerificationResult {
 	}
 }
 
-type PolicyOptionConfigurator func(*PolicyOptions) error
+type PolicyOption func(*PolicyConfig) error
+type ArtifactPolicyOption func(*PolicyConfig) error
+
+// PolicyBuilder is responsible for building & validating a PolicyConfig
+type PolicyBuilder struct {
+	artifactPolicy ArtifactPolicyOption
+	policyOptions  []PolicyOption
+}
+
+func (pc PolicyBuilder) Options() []PolicyOption {
+	arr := []PolicyOption{PolicyOption(pc.artifactPolicy)}
+	return append(arr, pc.policyOptions...)
+}
+
+func (pc PolicyBuilder) BuildConfig() (PolicyConfig, error) {
+	policy := &PolicyConfig{}
```

Comment:
Nit but if we are returning by value we should just do `policy := PolicyConfig{}` so we don't need to de-reference the `policy` when we return it below.